K epsilon model (#25752)
idaholab/moose: modules/navier_stokes/doc/content/source/fvkernels/INSFVTKESourceSink.md

# INSFVTKESourceSink

!syntax description /FVKernels/INSFVTKESourceSink

The object computes the turbulent source and sink term for the turbulent kinetic energy equation.

Two terms are computed `destruction` = $\epsilon$ and `production` = $G_k$ and the term $\epsilon - G_k$ is
passed to the residual.
A different treatment is used for the bulk and the near wall regions.

> *Bulk formulation*:

Author: Thanks

The turbulent production $G_k$ is modeled as:

\begin{equation}
G_k = \mu_t S^2 \,,
\end{equation}

where:

- $\mu_t$ is the turbulent dynamic viscosity,
- $S$ is the shear strain tensor internal norm, defined as $S = \sqrt{2\mathbf{S}:\mathbf{S}}$ with the shear strain tensor defined as $\mathbf{S} = \frac{1}{2} [\nabla \vec{u} + (\nabla \vec{u})^T]$.

The turbulent kinetic energy dissipation rate $\epsilon$ is generally coming from a coupled
transport equation for $\epsilon$.
However, for canonical or meassured cases, e.g., isotropic decaying turbulece,
the user can force its value to a functor.

> *Wall formulation*:

Author: Thanks

All cells in contact with a boundary identified in the `walls` list are applied a different
treatment for production and destruction.

> A separate formulation is used for the `sub-laminar` and `logarithmic` boundary layers.

Author: Tx

incremental fixed-point search algorithm.
The cells belonging to the `sub-laminar` boundary layers are defined as those
for which $y^+ < 11.25$.
The ones belonging to the `logarithmic` boundary layer are those for which $y^+ \ge 11.25$.

For both the `sub-laminar` and `logarithmic` boundary layers the production term is defined as:

\begin{equation}
G_k = C_{\mu}^{0.25} \mu_t \frac{k^{\frac{3}{2}} ||\nabla \vec{u}||}{\kappa y_p} \,,
\end{equation}

where:

- $C_{\mu} = 0.09$ is a closure parameter,
- $k$ is the turbulent kinetic energy,
- $||\nabla \vec{u}||$ is the near wall velocity gradient norm, which is defined as $$||\nabla \vec{u}||$ = (\nabla \vec{u} \cdot \hat{n}) \cdot \hat{n}$,
- $\kappa = 0.41$ is the von Kármán constant.

The formulation assumes that the nert wall value are already imposed in the $\mu_t$ functor.

When solving a linear problem, instead of the nonlinear formulation, the production term is formulated as:

\begin{equation}
G_k = C_{\mu}^{0.25} \mu_t \frac{k \sqrt{k_{old}}||\nabla \vec{u}||}{\kappa y_p} \,.
\end{equation}

where:

- $k_{old}$ is the value of the turbulent kinetic energy in the previous iteration.

> Whereas the linear or nonlinear formulation is used is controled by the `linearized_model` parameter.

Author: Thanks

# INSFVTKESourceSink

> !syntax description /FVKernels/INSFVTKESourceSink
> 

Reviewer: ```suggestion
```

The object computes the turbulent source and sink term for the turbulent kinetic energy equation.

Two terms are computed `destruction` = $\epsilon$ and `production` = $G_k$ and the term $\epsilon - G_k$ is
passed to the residual.
A different treatment is used for the bulk and the near wall regions.

## Bulk formulation:

The turbulent production $G_k$ is modeled as:

\begin{equation}
G_k = \mu_t S^2 \,,
\end{equation}

where:

- $\mu_t$ is the turbulent dynamic viscosity,
- $S$ is the shear strain tensor internal norm, defined as $S = \sqrt{2\mathbf{S}:\mathbf{S}}$ with the shear strain tensor defined as $\mathbf{S} = \frac{1}{2} [\nabla \vec{u} + (\nabla \vec{u})^T]$.

The turbulent kinetic energy dissipation rate $\epsilon$ is generally coming from a coupled
transport equation for $\epsilon$.
However, for canonical or meassured cases, e.g., isotropic decaying turbulece,
the user can utilize predefined fields through functors in MOOSE.

## Wall formulation:

> All cells in contact with a boundary identified in the `walls` list are applied a different

Reviewer: ```suggestion
All cells in contact with a boundary identified in the [!param](/FVKernels/INSFVTKESourceSink/walls) list are applied a different
```

incremental fixed-point search algorithm.
The cells belonging to the `sub-laminar` boundary layers are defined as those
for which $y^+ < 11.25$.
The ones belonging to the `logarithmic` boundary layer are those for which $y^+ \ge 11.25$.

For both the `sub-laminar` and `logarithmic` boundary layers the production term is defined as:

\begin{equation}
G_k = C_{\mu}^{0.25} \mu_t \frac{k^{\frac{3}{2}} ||\nabla \vec{u}||}{\kappa y_p} \,,
\end{equation}

where:

- $C_{\mu} = 0.09$ is a closure parameter,
- $k$ is the turbulent kinetic energy,
- $||\nabla \vec{u}||$ is the near wall velocity gradient norm, which is defined as $$||\nabla \vec{u}||$ = (\nabla \vec{u} \cdot \hat{n}) \cdot \hat{n}$,
- $\kappa = 0.41$ is the von Kármán constant.

The formulation assumes that the near wall value is already imposed in the $\mu_t$ functor.

When solving a linear problem, instead of the nonlinear formulation, the production term is formulated as:

\begin{equation}
G_k = C_{\mu}^{0.25} \mu_t \frac{k \sqrt{k_{old}}||\nabla \vec{u}||}{\kappa y_p} \,.
\end{equation}

where:

- $k_{old}$ is the value of the turbulent kinetic energy in the previous iteration.

> Whereas the linear or nonlinear formulation is used is controled by the

Reviewer: ```suggestion
Whether the linear or nonlinear formulation is used can be controlled by the
```

[!param](/FVKernels/INSFVTKESourceSink/linearized_model) parameter.

> For the destruction formulation is different for the `sub-laminar` and `logarithmic` layers.

Reviewer: ```suggestion
For the destruction, formulation is different for the `sub-laminar` and `logarithmic` layers.
```
